src/main/java/com/bravo/user/dao/repository/PaymentRepository.java

```diff
@@ -0,0 +1,13 @@
+package com.bravo.user.dao.repository;
+
+import com.bravo.user.dao.model.Payment;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+
+import java.util.List;
+
+public interface PaymentRepository extends JpaRepository<Payment, String>, JpaSpecificationExecutor<Payment> {
+    Page<Payment> findAllByUserId(String userId, Pageable pageable);
```

Comment:
How is this different than `findByUserId`?  Is `JpaSpecificationExecutor` actually required here?

Reply:
I created a custom method to add Pagination to the request, the last 3 years I have been doing Middleware BFF work and haven't used JPA as we are calling existing services and aggregating then for the client, so this would be where I have less experience.

I dont think it is, looks like I may have left that in when I was modeling this after the User resource.